lib/widget/video.dart

```diff
@@ -23,42 +26,102 @@ class Video extends StatefulWidget
 
   @override
   Map<String, Function> getters() {
-    return {};
+    return {
+      'isInitialized': () => _controller._playerController?.value.isInitialized,
+      'isBuffering': () => _controller._playerController?.value.isBuffering,
+      'isLooping': () => _controller._playerController?.value.isLooping,
+      'isPlaying': () => _controller._playerController?.value.isPlaying,
+      'isEnded': () => _controller._playerController?.value.isCompleted,
+    };
   }
 
   @override
   Map<String, Function> methods() {
-    return {};
+    return {
+      'stop': () => _controller.playerCapabilities?.stop(),
+      'pause': () => _controller.playerCapabilities?.pause(),
+      'play': () => _controller.playerCapabilities?.play(),
+      'mute': () => _controller.playerCapabilities?.mute(),
+      'unmute': () => _controller.playerCapabilities?.unmute(),
+      'seekTo': (value) => _controller.playerCapabilities?.seekTo(value),
+      'setPlaybackRate': (value) =>
+          _controller.playerCapabilities?.setPlaybackRate(value),
+    };
   }
 
   @override
   Map<String, Function> setters() {
     return {
       'source': (value) => _controller.updateSource(Utils.getUrl(value)),
+      'showControls': (value) =>
+          _controller.showControls = Utils.getBool(value, fallback: true),
+      'loadingWidget': (loadingWidget) =>
+          _controller.loadingWidget = loadingWidget,
+      'enableLoop': (value) => _controller._playerController
+          ?.setLooping(Utils.getBool(value, fallback: false)),
+      'autoplay': (value) =>
+          controller.autoplay = Utils.getBool(value, fallback: false),
+      'playbackRate': (value) =>
+          controller.playbackRate = Utils.optionalDouble(value, max: 2.0),
+      'volume': (value) =>
+          controller.volume = Utils.optionalDouble(value, max: 1.0),
+      'onChange': (definition) => _controller.onChange =
+          EnsembleAction.fromYaml(definition, initiator: this),
+      'onStart': (definition) => _controller.onStart =
+          EnsembleAction.fromYaml(definition, initiator: this),
+      'onEnd': (definition) => _controller.onEnd =
+          EnsembleAction.fromYaml(definition, initiator: this),
     };
   }
 }
 
 class MyController extends WidgetController {
+  PlayerCapabilities? playerCapabilities;
+  EnsembleAction? onStart;
+  EnsembleAction? onChange;
+  EnsembleAction? onEnd;
+  EnsembleAction? onComplete;
```

Comment:
not exposed in schema.

Reply:
onComplete is not used in the code. Just removed it.